src/optimagic/logging/logger.py

```diff
@@ -0,0 +1,320 @@
+        return exploration
+
+    @staticmethod
+    def _extract_best_history(
+        histories: pd.DataFrame, optimization_type: OptimizationType
+    ) -> tuple[IterationHistory, list[IterationHistory] | None]:
+        groupby_step_criterion = histories["criterion"].groupby(level="step")
+
+        if optimization_type is OptimizationType.MINIMIZE:
+            best_idx = groupby_step_criterion.min().idxmin()
+        else:
+            best_idx = groupby_step_criterion.max().idxmax()
+
+        remaining_indices = (
+            histories.index.get_level_values("step").unique().difference([best_idx])
+        )
+
+        best_history: pd.DataFrame | pd.Series[Any] = histories.xs(
+            best_idx, level="step"
+        )
+
+        def _to_dict(pandas_obj: pd.DataFrame | pd.Series) -> dict[str, Any]:  # type:ignore
+            if isinstance(pandas_obj, pd.DataFrame):
+                result = pandas_obj.to_dict(orient="list")
+            else:
+                result = best_history.to_dict()
+            return cast(dict[str, Any], result)
+
+        best_history_dict = _to_dict(best_history)
+        local_histories = [
+            _to_dict(histories.xs(idx, level="step")) for idx in remaining_indices
+        ]
+        if len(local_histories) == 0:
+            remaining_histories = None
+        else:
+            remaining_histories = [
+                IterationHistory(**history_dict) for history_dict in local_histories
+            ]
+
+        return IterationHistory(**best_history_dict), remaining_histories
+
+    def read_multistart_history(
+        self, direction: OptimizationType | OptimizationTypeLiteral
+    ) -> MultiStartIterationHistory:
+        """Read and the multistart optimization history.
+
+        Args:
+            direction: The optimization direction, either as an enum or string.
+
+        Returns:
+            A `MultiStartIterationHistory` object containing the best history,
+                local histories, and exploration history.
+
+        """
+        optimization_type = self._normalize_direction(direction)
+        history_df = self._build_history_dataframe()
+        exploration, optimization_history = self._split_exploration_and_optimization(
+            history_df
+        )
+        exploration_history = self._sort_exploration(exploration, optimization_type)
+        best_history, remaining_histories = self._extract_best_history(
+            optimization_history, optimization_type
+        )
+
+        return MultiStartIterationHistory(
+            best_history,
+            local_histories=remaining_histories,
+            exploration=exploration_history,
+        )
+
+    def read_start_params(self) -> PyTree:
+        """Read the start parameters form the problem store.
+
+        Returns:
+            A pytree object representing the start parameter.
+
+        """
+        return self.problem_store.select_last_rows(1)[0].params
```

Comment:
Why does `select_last_rows(1)` return the start params? The start params should be the first, not the last entry.

Reply:
Agree, will do